lib/rspec/rails/example/mailbox_example_group.rb

```diff
@@ -0,0 +1,74 @@
+module RSpec
+  module Rails
+    # @api public
+    # Container module for mailbox spec functionality.
+    module MailboxExampleGroup
+      extend ActiveSupport::Concern
+
+      if RSpec::Rails::FeatureCheck.has_action_mailbox?
+        require 'action_mailbox/test_helper'
+        extend ::ActionMailbox::TestHelper
```

Comment:
I don't love this, but `ActionMailbox` appears to expect to operate on a particular named `ActionMailbox::InboundEmail < ActiveRecord::Base`, and I'd be worried that using some kind of double here might be brittle.

Reply:
Why don't you like it?

Reply:
This is here mostly because [`ActionMailbox::InboundEmail.create_and_extract_message_id!` is doing a lot of work](https://github.com/rails/rails/blob/3d7391cb76758c82c8ea7759b6dd8d4402bc2346/actionmailbox/app/models/action_mailbox/inbound_email/message_id.rb#L16) that I didn't want to duplicate here, but this feels like I'm importing implementation details, not clean abstractions.

My inclination here is to do something like

* define a lightweight e.g. `RSpec::Rails::ActionMailbox::MockInboundEmail` that we use internally in `process_inbound_email`
* add shared examples for `::ActionMailbox::InboundEmail` and `MockInboundEmail` to make sure they are interchangeable
* consider using `MockInboundEmail` for the `process` helper (either by default or opt-in somehow?)
* work on landing `MockInboundEmail` upstream

How does that sound to y'all?

Reply:
I love the idea of a `MockInboundEmail` but I think it can be another PR?